maven_iuvs/echelle.py

```diff
@@ -2630,43 +2697,102 @@ def line_fit_initial_guess(wavelengths, spectrum, H_a=20, H_b=170, D_a=80, D_b=1
     ----------
     Vector of initial guess values
     """
+    num_params = 8
+
     # Total flux of H and D in DN, initial guess: get by integrating around the line. Note that the H bounds as defined
     # in a parent function overlap the D, but that's okay for an initial guess.
+    rows = get_n_int(light_fits) if coadded is False else 1
+    guesses = np.zeros((rows, num_params))
+    # Line center of IPH is predicted based on the obs geometry;
+    # length is number of integrations.
+    lambda_IPH_lya_guess = 121.567 + predict_IPH_linecenter(light_fits)
```

Comment:
probably should fit the IPH parameter the same way as D, by fitting the H central wavelength and then subtracting the IPH line center offset from the lab wavelength, but if this works that's fine for now.

Reply:
let me know if you're planning on making this change or just leaving it as is.

Reply:
We keep it because it represents the uncertainty in the offset relative to the mars rest wavelength. The issue is whether a "0" fit parameter for the IPH offset within the bounds of +/-5 km/s represents zero offset from the lab frame wavelength of Lyman alpha or from the fitted instrument Mars H Ly a wavelength. I think it should be the latter to minimize potential systematics involved in referencing the IPH to the lab wavelength but fitting the observed location of the Mars H line center.

Reply:
After discussion, I implemented this in 6795164877bbaa7fd02e2528f26cfc204b03e771, but found that it adds a huge amount of overhead to the fitting for some reason - 5-10 seconds per call, which is not sustainable. Thus I've reverted it in 4fa5e9b4d53c3d383802df8f5c6abeb51708496c and for now we'll just accept that there will be a slight discrepancy that should be accounted for within the uncertainties anyway.